From d63f7434c09a056916de95ab9dffd540f450e2b1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 10 Nov 2022 11:38:45 +0200
Subject: [PATCH] Revert "[SW-105874] gaudi2: remove usage of arc1 dccm for arc0"

This reverts commit 3772e76338f9febed78da379a5253f2f6d512f3c.

It is observed with certain setups, pcie link is not established
on sudo reboot with ARC0 using its in own DCCM. Revert this change
till issue is analyzed.

Change-Id: Ifd932558576ce568dee20999cf9d60f3d0845e44
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 28b322c..9f220a9 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -49,7 +49,6 @@
 #include <gaudi2/arc/dev_security.h>
 #include <tpm2.h>
 #include <errno.h>
-#include <device.h>
 
 #define PRODUCT_ID_STR_SIZE			0x8
 
@@ -705,25 +704,8 @@
 	sleep();
 }
 
-static void secondary_device_init(void)
-{
-	/*
-	 * Perform data structure initialization only for the
-	 * global structures which are used by ARC0 for PCIe
-	 * bring up sequence.
-	 * Note: If any additional functionality is added
-	 * in PCIe routine which make use of new global data
-	 * structure, should be initialized here before use.
-	 */
-	device_init_efuse();
-	device_init_pci();
-	device_init_pll();
-	udelay_init();
-}
-
 void secondary_main(void)
 {
-	secondary_device_init();
 	preboot_pcie_monitoring();
 
 	/* Put secondary CPU in sleep state until further notice */
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 4b80268..cbf43cb 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -80,7 +80,7 @@
  * Let PPBoot start from the offset of 0x4030, for using DCCM
  * for its own BSS and Stack.
  */
-#define PPBOOT_DCCM_BASE	0x80004030
+#define PPBOOT_DCCM_BASE	0x27064030
 #define PPBOOT_DCCM_SZ		0x2000	/* 8KB */
 
 #define SRAM_BIST_DATA_BASE	(PPBOOT_DCCM_BASE + PPBOOT_DCCM_SZ)
diff --git a/arch/gaudi2/arc/secondary.S b/arch/gaudi2/arc/secondary.S
index cef77bb..2c0e567 100644
--- a/arch/gaudi2/arc/secondary.S
+++ b/arch/gaudi2/arc/secondary.S
@@ -7,28 +7,12 @@
 
 .global secondary_entry
 secondary_entry:
-	mov	sp, __stack_top
+	/* stack top set to secondary CPU DCCM 1k offset (internal address) */
+	mov	sp, 0x80000400
 	mov	fp, sp
 
 	/* Disable un-aligned checks on data memory */
 	lr r0, [STATUS32]
 	bset r0, r0, 0x13
 	kflag r0
-
-	/*
-	 * ARC0 DCCM init is performed by ppboot code running on ARC1
-	 * before putting ARC0 in running state.
-	 * copy data and sdata sections from FWSRAM to ARC0 DCCM
-	 */
-	mov_s   r1, @_erodata           ; r1 = end of the text/ro section
-	mov_s   r2, @_data
-	sub     r3, @_esdata, r2        ; r3 = size of total data in bytes
-
-	asr_s   r13, r3
-	asr.f   lp_count, r13
-
-	lpnz    1f
-	ld.ab   r5,[r1,4]
-	st.ab   r5,[r2,4]
-1:
 	bl	secondary_main
